app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1271,7 +1271,7 @@ public static void createPreferencesMenu(Activity activity) {
     }
 
     private void startAdvancedActionsActivity() {
-        startActivity(new Intent(this, AdvancedActionsActivity.class));
+        startActivityForResult(new Intent(this, AdvancedActionsActivity.class), ADVANCED_ACTIONS_ACTIVITY);
```

Comment:
the primary issue was that `AdvancedActionsActivity` was not being started for result, so the `handleAdvancedActionResult` callback was never being called.